### ISSUE TITLE
Redacted m.room.topic event is still used to set the topic

### ISSUE THREAD

**Issue by KitsuneRal** [vector-im/element-web#7683]
The problem is two-fold:
1. When I redact the topic event from Riot, it doesn't actually do anything to the topic.
2. ~~When I open Riot and the last `m.room.topic` event coming from the sync is redacted, Riot still uses it to set (blank) topic instead of finding the previous topic.~~ (see below)

### Steps to reproduce

Case 1:
0. Pre-requisites: Riot running, room opened, topic set
1. Redact the last `m.room.topic` event in the timeline.
2. Riot keeps showing the topic as if nothing happened. Inspecting in `/devtools` shows that the event has been redacted though.

Case 2 (retracted, see [the comment](https://github.com/vector-im/riot-web/issues/7683#issuecomment-439605097)):
~~0. Pre-requisites: Riot doesn't run, in another client there's a room with the non-empty topic.~~
~~1. In another client, change the topic and immediately redact the change.
2. Start Riot, open the same room.
3. Riot shows the empty topic instead of the previous topic. Inspecting in `/devtools` shows the redacted `m.room.topic`.~~

### Version information

- **Platform**: web
- **Browser**: Firefox
- **OS**: Windows 10
- **URL**: riot.im/develop

**Comment by KitsuneRal** [comment]
TL;DR: the CS API spec supports Case 1 and says nothing on Case 2; Case 2 is rebuked by the SS API spec, though, so I'm retracting it.
Case 1: The Spec says that redaction "Strips all information out of an event which isn't critical to the integrity of the server-side representation of the room", and "when a client receives a redaction event it should change the redacted event in the same way a server does". I believe this kinda implies that the client should erase the redacted content locally if the user redacts the event. Topic, in particular, is by no means critical to the integrity of the server-side representation.
Case 2: This is trickier; apparently, as discussed with @ara4n, Riot is correct in showing the empty topic in this case, since this is the actual state of the room, rather than (as I incorrectly assumed) the previous topic. CS API spec is not clear on this: while it mentions that, e.g., membership state should remain intact along with event type/state key, it says nothing about (non-)reverting the room state upon a redaction. The Federation API spec does quickly mention in a note when discussing [event rejection](https://matrix.org/docs/spec/server_server/unstable.html#rejection) that redaction events, unlike rejected events, "can still affect the state of the room".

**Comment by turt2live** [comment]
We fixed this.
